diff --git a/Inc/stm32l4xx_hal_conf.h b/Inc/stm32l4xx_hal_conf.h
index 9dd6089..4ef38d6 100644
--- a/Inc/stm32l4xx_hal_conf.h
+++ b/Inc/stm32l4xx_hal_conf.h
@@ -122,6 +122,18 @@
 #endif /* HSI_VALUE */
 
 /**
+  * @brief Internal High Speed oscillator (HSI48) value for USB FS, SDMMC and RNG.
+  *        This internal oscillator is mainly dedicated to provide a high precision clock to
+  *        the USB peripheral by means of a special Clock Recovery System (CRS) circuitry.
+  *        When the CRS is not used, the HSI48 RC oscillator runs on it default frequency
+  *        which is subject to manufacturing process variations.
+  */
+#if !defined  (HSI48_VALUE) 
+ #define HSI48_VALUE   ((uint32_t)48000000U) /*!< Value of the Internal High Speed oscillator for USB FS/SDMMC/RNG in Hz.
+                                              The real value my vary depending on manufacturing process variations.*/
+#endif /* HSI48_VALUE */
+
+/**
   * @brief Internal Low Speed oscillator (LSI) value.
   */
 #if !defined  (LSI_VALUE) 
diff --git a/Makefile b/Makefile
index 56e03c6..c75b196 100644
--- a/Makefile
+++ b/Makefile
@@ -58,7 +58,7 @@ Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_pwr_ex.c \
 
 # ASM sources
 ASM_SOURCES =  \
-startup_stm32l476xx.s
+startup_stm32l432xx.s
 
 
 #######################################
@@ -96,7 +96,7 @@ AS_DEFS =
 # C defines
 C_DEFS =  \
 -DUSE_HAL_DRIVER \
--DSTM32L476xx
+-DSTM32L432xx
 
 
 # AS includes
@@ -130,7 +130,7 @@ CFLAGS += -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)"
 # LDFLAGS
 #######################################
 # link script
-LDSCRIPT = STM32L476RGTx_FLASH.ld
+LDSCRIPT = STM32L432KCUx_FLASH.ld
 
 # libraries
 LIBS = -lc -lm -lnosys
